### PR TITLE
[Feature](add bitmap udaf) add the bitmap intersection and difference set for mixed calculation of udaf

### PR DESCRIPTION
**Pull request by zhbinbin**
## Problem summary
Add the bitmap set for  mixed calculation of udaf
Calculate the bitmap intersection, union and difference set of expressions to calculate the count function.

## Checklist(Required)
1. Does it affect the original behavior: 
    - [No]
2. Has unit tests been added:
    - [No]
3. Has document been added or modified:
    - [Yes]
4. Does it need to update dependencies:
    - [No ]
5. Are there any changes that cannot be rolled back:
    - [ No]

## Further comments
If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.33 seconds
 load time: 649 seconds
 storage size: 17123449725 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230105032640_clickbench_pr_73859.html

File: be/src/util/bitmap_expr_calculation.h [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+        }
+
+        while (!opStack.empty()) {
+            polish.push(opStack.top());
+            opStack.pop();
+        }
+        return printStack(polish);
+    }
+
+    // bitmap交并差运算 因为数据是放在堆栈中 所以前一个操作数是opB 后一个操作数是opA
```
Comment (yangzhg) [inline be/src/util/bitmap_expr_calculation.h:204]: translate to en

File: be/src/util/bitmap_intersect.h [hunk 2]
```diff
@@ -84,7 +84,7 @@ inline char* Helper::write_to<StringRef>(const StringRef& v, char* dest) {
 
 template <>
 inline char* Helper::write_to<std::string>(const std::string& v, char* dest) {
-    *(uint32_t*)dest = v.size();
+    *(int32_t*)dest = v.size();
```
Comment (yangzhg) [inline be/src/util/bitmap_intersect.h:87]: int32 maybe overflow

File: be/src/util/bitmap_expr_calculation.h [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+// Example: orthogonal_bitmap_expr_calculate(user_id, event, '(A|B)&(C-D)'), meaning find the intersection union difference set of user_id in all A/B/C/D 4 bitmaps
+// Operation symbol:
+// the operator '|' stands for union, the operator '&' stands for intersection, the operator '-' indicates the difference set, the operator '^' stands for xor
+class BitmapExprCalculation : public BitmapIntersect<std::string> {
+public:
+    BitmapExprCalculation() = default;
+
+    explicit BitmapExprCalculation(const char* src) { deserialize(src); }
+
+    void bitmap_calculation_init(std::string& inputStr) {
```
Comment (yangzhg) [inline be/src/util/bitmap_expr_calculation.h:36]:
```suggestion
    void bitmap_calculation_init(std::string& input_str) {
```

File: be/src/util/bitmap_expr_calculation.h [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    // calculate the bitmap value by expr bitmap calculate
+    int64_t bitmap_calculate_count() {
+        if (_bitmaps.empty()) {
+            return 0;
+        }
+        return bitmap_calculate().cardinality();
+    }
+
+private:
+    int priority(char c) {
```
Comment (yangzhg) [inline be/src/util/bitmap_expr_calculation.h:110]:
```suggestion
    constexpr int priority(char c) {
```

File: be/src/util/bitmap_expr_calculation.h [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+            return 1;
+        case '-':
+            return 1;
+        default:
+            return 0;
+        }
+    }
+
+    template <class T>
+    std::string printStack(std::stack<T> stack) {
```
[the remark this answers is not in the document]
Comment (zhbinbin) [reply be/src/util/bitmap_expr_calculation.h]: remove const , pop has change

File: be/src/util/bitmap_expr_calculation.h [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+            return 0;
+        }
+    }
+
+    template <class T>
+    std::string print_stack(const std::stack<T>& stack) {
+        std::string result;
+        while (!stack.empty()) {
+            result = stack.top() + result;
+            stack.pop();
```
[the remark this answers is not in the document]
Comment (zhbinbin) [reply be/src/util/bitmap_expr_calculation.h:130]: good

**Comment by yangzhg** [comment]
run buildall

**Comment by chenlinzhong** [comment]
run p0

File: docs/zh-CN/docs/advanced/orthogonal-bitmap-manual.md [hunk 7]
```diff
@@ -161,3 +191,10 @@ orthogonal_bitmap_union_count(bitmap_column)
 ```sql
 select orthogonal_bitmap_union_count(user_id) from user_tag_bitmap where tag in (13080800, 11110200);
 ```
+
+bitmap交并差集合混合计算：
+
+```sql
+select orthogonal_bitmap_expr_calculate_count(user_id, tag, '(A|B)&(C-D)&E') from user_str_tag_bitmap where tag in ('A', 'B', 'C', 'D', 'E');
```
Comment (chenlinzhong) [inline docs/zh-CN/docs/advanced/orthogonal-bitmap-manual.md:198]: how to deal with tag with "& | -"

**Comment by chenlinzhong** [comment]
run beut

**Comment by chenlinzhong** [comment]
run feut

**Comment by chenlinzhong** [comment]
run p0

**Comment by chenlinzhong** [comment]
run buildall

**Comment by chenlinzhong** [comment]
run buildall

**Comment by chenlinzhong** [comment]
run buildall

**Comment by chenlinzhong** [comment]
run p0

**Comment by chenlinzhong** [comment]
run p0

**Comment by chenlinzhong** [comment]
run p0

**Review by chenlinzhong — APPROVED** [review]
LGTM

**Comment by yuxuan-luo** [comment]
@morningman

**Pull request merged**